packages/schema-blocks/src/blocks/RequiredBlocks.tsx

```
@@ -0,0 +1,81 @@
+import { ReactElement } from "react";
+import { BlockInstance, createBlock } from "@wordpress/blocks";
+import { createElement } from "@wordpress/element";
+import { RequiredBlock } from "../instructions/blocks/dto";
+
+import { getInnerblocksByName, insertBlockToInnerBlock } from "../functions/innerBlocksHelper";
+import { getBlockType } from "../functions/BlockHelper";
+import { PanelBody } from "@wordpress/components";
+
+/**
+ * Renders a list with the required block names and an button to add/remove one.
+ *
+ * @param {BlockInstance}   block          The block to render the list for.
+ * @param {RequiredBlock[]} requiredBlocks The required blocks.
+ *
+ * @returns {ReactElement} The rendered block.
+ */
+export default function RequiredBlocks( block: BlockInstance, requiredBlocks: RequiredBlock[] ): ReactElement {
+	// Retrieve a list with names.
+	const requiredBlockNames = requiredBlocks.map( ( requiredBlock ) => {
+		return requiredBlock.name;
+	} );
+
+	const findPresentBlocks = getInnerblocksByName( block, requiredBlockNames );
+	const presentBlockNames = findPresentBlocks.map( ( presentBlock ) => {
+		return presentBlock.name;
+	} );
+
+	const requiredBlockItems: ReactElement[] = [];
+
+	requiredBlockNames.forEach( ( requiredBlockName: string, index: number ) => {
+		const blockType = getBlockType( requiredBlockName );
+
+		if ( typeof blockType === "undefined" ) {
+			return;
+		}
```

I am pondering whether we can remove this check...

Removing this would mean that we can transform this `forEach` + `push` loop into a `map`. Since a block is either missing or not. 

Though it can be worthwhile to do some defensive programming 🤔. 

Or is this possibly a case where we can apply exception handling? Since we expect a block type to be registered for a required block name. Perhaps failing silently, as we do here is not what we want, since we may want to indicate that something went wrong with an error message in the console.

Just my thoughts 😛. What do you think @andizer?